CMakeLists.txt

```diff
@@ -43,30 +56,62 @@ set(SERENITY_SOURCES
 )
 
 set(PIPELINE_TEST_SOURCES
-    pipeline/pipeline_test.cpp
+    tests/integration/pipeline_test.cpp
 )
 
-set(SERENITY_TEST_SOURCES
+set(SERENITY_UT_TESTS_SOURCES
+    tests/ut/estimators/estimator_test.cpp
+    tests/ut/controllers/qos_controller_test.cpp
+    tests/ut/filters/moving_average_test.cpp
     tests/main.cpp
-    tests/filters/moving_average_test.cpp
-    tests/estimator_test.cpp
-    tests/qos_controller_test.cpp
-    tests/serenity.hpp
+    tests/integration/sources/json_source_test.cpp
 )
 
+set(SERENITY_INTEGRATION_TESTS_SOURCES
+    tests/main.cpp
+    tests/integration/sources/json_source_test.cpp
+)
+
+set(SERENITY_TESTS_HELPERS_SOURCES
+    tests/helpers/sources/json_source.cpp
+)
+
+link_directories(${MESOS_INSTALLATION_DIR}/lib/)
+
+#Serenity library
 add_library(serenity ${SERENITY_SOURCES})
-target_link_libraries(serenity messages)
+target_link_libraries(serenity messages mesos glog)
 
-add_executable(pipeline_test ${PIPELINE_TEST_SOURCES})
-target_link_libraries(pipeline_test serenity mesos)
+#Integration tests library
+add_library(serenity-tests-helpers ${SERENITY_TESTS_HELPERS_SOURCES})
+target_link_libraries(serenity-tests-helpers serenity messages-test pbjson)
 
-add_executable(serenity-tests ${SERENITY_TEST_SOURCES})
+#Unit tests
+add_executable(serenity-tests ${SERENITY_UT_TESTS_SOURCES})
 target_link_libraries(serenity-tests
     serenity
+    serenity-tests-helpers
     gtest
     gtest_main
     gmock
     gmock_main
-    mesos
 )
 add_test(serenity-tests serenity-tests)
+
+
```

Comment:
kill line for consistency with this file (we use 1 newline to separate commands)